### PR TITLE
Update  block failed message 

### LINKED ISSUE #2414: Improve error message for pokladot failing to decode block
## Description

Currently when a block cannot be decoded the error message just says that it failed to index block without any indication how to solve the issue.

The error should be updated for substrate with the following message:
```
Failed to index block ${blockNum}. This is because the block cannot be decoded. To solve this you can either:
* Skip the block
* Update the chain types. You can test this by viewing the block with https://polkadot.js.org/apps/
For further information please read the docs: https://academy.subquery.network/
```

### PR DESCRIPTION
**Pull request by yoozo**
# Description
Update block failed message.

Fixes #2414 (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [ ] I have tested locally
- [x] I have performed a self review of my changes
- [ ] Updated any relevant documentation
- [ ] Linked to any relevant issues
- [ ] I have added tests relevant to my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] My code is up to date with the base branch
- [ ] I have updated relevant changelogs. [We suggest using chan](https://github.com/geut/chan/tree/main/packages/chan)

### CODE REVIEW

**Review by stwiname — COMMENTED** [review]
The updated error message should be for a specific error not all errors failing to index blocks. 

It needs to handle a specific error type here https://github.com/subquery/subql/blob/main/packages/node/src/utils/substrate.ts#L257

**Comment by jiqiang90** [comment]
It will also face decode error at https://github.com/subquery/subql/blob/3e1d2ee7ea1ddce0c48b71d9d77b329121edb433/packages/node/src/utils/substrate.ts#L332

**Comment by yoozo** [comment]
The comments have all been fixed.

**Review by stwiname — COMMENTED** [review]
Looking much better, it needs to catch a specific type of error though, not just all errors from these methods.

@jiqiang90 can you please help finding a reproducible case if needed

**Comment by jiqiang90** [comment]
> The updated error message should be for a specific error not all errors failing to index blocks.
> 
> It needs to handle a specific error type here https://github.com/subquery/subql/blob/main/packages/node/src/utils/substrate.ts#L257

I think from our previous experience, only these two calls when fetch block and get event will throw decode error, I'm happy with this change. 

@yoozo we can try this project , with change start height to 43249 and remove chainTypes. https://github.com/subquery/subql-starter/blob/main/Karura/karura-starter/project.ts

Will be nice we and include a unit test here too

**Review by jiqiang90 — APPROVED** [review]

File: packages/node/src/utils/substrate.test.ts [hunk 1]
```diff
@@ -65,4 +66,22 @@ describe('substrate utils', () => {
       filterExtrinsic(block.extrinsics[2], { isSigned: false }),
     ).toBeFalsy();
   });
+
+  it('decode fail message', async () => {
+    const provider = new WsProvider(ENDPOINT_KARURA);
+    const api = await ApiPromise.create({ provider });
+
+    await expect(getBlockByHeight(api, 86614)).rejects.toThrow(
+      /Unable to decode|failed decoding|unknown type/,
```
Comment (stwiname) [inline packages/node/src/utils/substrate.test.ts:75]: We want to test that it throws an error with our message and not the underlying error that we wrap

Comment (yoozo) [reply packages/node/src/utils/substrate.test.ts:75]: I think both are needed.

**Review by stwiname — APPROVED** [review]

**Pull request merged**